code/iLasso_for_SCINGE.m

```diff
@@ -20,47 +20,101 @@
+% Load expression matrix and pseudotime
+X = m.X;
+ptime = m.ptime;
+
+% Precompute the full kernel matrix -- to be subsampled for each regulator
+if m.computeKp
+    if ismember('fullKp',who(m))&&~isempty(m.fullKp)
+        m.fullKp(1,:) = [];
+    end
+    Kp2.Kp = (zeros(length(ptime)));
+    refT = (ptime);
+    for k = 1:L
+        Kp = (refT-(L)*Dt+(k-1)*Dt);
+        Kp = bsxfun(gkern,Kp',refT);
+        Kp2(k).Kp = (Kp);
+        m.fullKp(1,k) = Kp2(k);
+    end
+    m.computeKp = 0;
+    clear Kp Kp2;
+end
+
+X1 = m.X1;
+tind = find(X1(pa(1),:));
+tval = X(pa(1),:);
+%target{1}(:,tind) = [];
+ttime = ptime;
+ttime(tind) = [];
+tval(tind) = [];
+B = sum(ttime<=(L*Dt));
+N1 = size(ttime, 2);
 
 % Build the matrix elements
-Am = (zeros(N1-B, P*L));
-bm = (zeros(N1-B, 1));
+Am = (zeros(N1-B, numregs*L));
 % Building the design matrix
-for j = 1:P
-    for i = (B+1):N1
-        bm(i-B) = Series{1}(1, i);
-        ti = (Series{1}(2, i) - (L)*Dt):Dt:(Series{1}(2, i)-Dt);
-   %     ti = repmat(ti, length(Series{j}(2, :)), 1);
- %       tSelect = repmat(Series{j}(2, :)', 1, L0);
-        tSelect = Series{j}(2, :)';
-        %ySelect = repmat(Series{j}(1, :)', 1, L0);
-        ySelect = Series{j}(1, :)';
+for k = 1:L
+    Kp2 = m.fullKp(1,k);
+    for j = 1:numregs
+        %tSelect = repmat(Series{j}(2, :)',1,L0);
+        rind = find(X1(pa(j),:));
+        ySelect = (full(X(pa(j),:)'));
+        refL = length(ySelect);
+        tSelect = ptime;
+        ySelect(rind) = [];
+        tSelect(rind) = [];
+        refind = 1:refL;
+        remind = refind(~(ismember(refind,tind))); % setdiff(refind,tind);
+        remind = union(1:remind(B+1)-1,tind);
+        rind = refind(~(ismember(refind,rind)));
+        remind = refind(~(ismember(refind,remind)));
+        %Kp = Kp2(1).Kp;
+        %    for i = (B+1):N1
+        %bm(i-B) = Series{1}(1, i);
+        % ti = ones(length(Series{j}(2, :)), 1)*((Series{1}(2, i) - (L)*Dt):Dt:(Series{1}(2, i)-Dt+Dt*params.AllowZeroLag));
+        %ti = repmat((Series{1}(2, i) - (L)*Dt):Dt:(Series{1}(2, i)-Dt+Dt*params.AllowZeroLag),length(tSelect),1);
+        %     Kp = (ttime(1, B+1:end)' - (L)*Dt)+(k-1)*Dt;
         switch krnl
             case 'Sinc'     % The sinc Kernel
                 Kp = sinc((ti-tSelect)/SIG);
             case 'Dist'     % The Dist Kernel
                 Kp = SIG./((ti-tSelect).^2);
             otherwise
-                Kp = exp(-((ti-tSelect).^2)/SIG);        % The Gaussian Kernel
+                % tic;
+                % Kp1 = bsxfun(gkern,Kp,tSelect);        % The Gaussian Kernel
+                % toc
+                %               Kp(:,rind) = [];
+                %              Kp(remind,:) = [];
         end
-        Am(i-B, ((j-1)*L0+1):(j*L0)) = (ySelect'*Kp)./sum(Kp);
+        %    Am(:, ((j-1)*L+k)) = (Kp2(k).Kp(keepind,rind1)*ySelect)./sum(Kp2(k).Kp(keepind,rind1),2);
+        Am(:, ((j-1)*L+k)) = (Kp2.Kp(remind,rind)*ySelect)./sum(Kp2.Kp(remind,rind),2);
```

Comment:
`k` corresponds to each lag in the GLG coefficient matrix.  Compute columns of `A` for each value of possible lags.  Can keep one kernel in memory at a time.